services/impl/src/main/java/org/exoplatform/automatic/translation/impl/AutomaticTranslationServiceImpl.java

```
@@ -80,30 +83,39 @@ public boolean setActiveConnector(String name) {
       LOG.error("Try to change automatic translation connector but feature is not active");
       return false;
     }
-    //should return false if connector name do not exists
-    //but not if the name is empty (correspond to unset case)
+    // should return false if connector name do not exists
+    // but not if the name is empty (correspond to unset case)
     if (name == null) {
       name = "";
     }
     String finalName = name;
-    if (finalName.isBlank() ||
-        translationConnectors.keySet().stream().anyMatch(connectorName -> connectorName.equals(finalName))) {
+    if (finalName.isBlank()
+        || translationConnectors.keySet().stream().anyMatch(connectorName -> connectorName.equals(finalName))) {
       settingService.set(Context.GLOBAL, Scope.GLOBAL, AUTOMATIC_TRANSLATION_ACTIVE_CONNECTOR, new SettingValue<>(name));
       return true;
     } else {
-      LOG.error("Try to change automatic translation connector with a non existing connector name : {}",name);
+      LOG.error("Try to change automatic translation connector with a non existing connector name : {}", name);
       return false;
     }
   }
 
   @Override
   public boolean setApiKey(String connector, String apikey) {
     AutomaticTranslationComponentPlugin connectorPlugin = translationConnectors.get(connector);
-    if (connectorPlugin==null) {
+    if (connectorPlugin == null) {
       return false;
     } else {
       return connectorPlugin.setApiKey(apikey);
     }
   }
 
+  @Override
+  public String translate(String message, Locale targetLang) {
+    String activeConnector = getActiveConnector();
+    if (activeConnector == null) {
+      return null;
+    }
+    return translationConnectors.get(activeConnector).translate(message, targetLang);
```

please add a null check, if null, it would be better to throw a runtime exception with a clear message : 'connector not found'

---

done